File: _Collection_of_Problems/Backjoon/8979/main.py
"""
첫번재 풀이
"""
import sys
from collections import defaultdict

if __name__ == "__main__":
    input = sys.stdin.readline
    N, K = map(int, input().split())

    medal_to_nation = defaultdict(set)
    for _ in range(N):
        nation, gold, silver, bronze = map(int, input().split())
        medal = (gold, silver, bronze)
        medal_to_nation[medal].add(nation)

    sorted_items = sorted(medal_to_nation.items(), reverse=True)

    rank = 1
    for _, nations in sorted_items:
        if K in nations:
            print(rank)
            break
        rank += len(nations)

"""
두번재 풀이
"""
import sys

if __name__ == "__main__":
    input = sys.stdin.readline
    N, K = map(int, input().split())

    countries = []
    for _ in range(N):
        nation, gold, silver, bronze = map(int, input().split())
        countries.append((gold, silver, bronze, nation))

    countries.sort(key=lambda x: (x[0], x[1], x[2]), reverse=True)

    k_medals = ()
    for g, s, b, n in countries:
        if n == K:
            k_medals = (g, s, b)
            break
    
    rank = 0
    for i in range(N):
        if (countries[i][0], countries[i][1], countries[i][2]) == k_medals:
            rank = i + 1
            break
            
    print(rank)

File: _Collection_of_Problems/Backjoon/8979/Main.java
import java.io.*;
import java.util.*;

class Country {
    int id;
    int gold;
    int silver;
    int bronze;

    public Country(int id, int gold, int silver, int bronze) {
        this.id = id;
        this.gold = gold;
        this.silver = silver;
        this.bronze = bronze;
    }
}

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        List<Country> countries = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            int id = Integer.parseInt(st.nextToken());
            int gold = Integer.parseInt(st.nextToken());
            int silver = Integer.parseInt(st.nextToken());
            int bronze = Integer.parseInt(st.nextToken());
            countries.add(new Country(id, gold, silver, bronze));
        }

        Country targetCountry = null;
        for (Country c : countries) {
            if (c.id == K) {
                targetCountry = c;
                break;
            }
        }

        int rank = 1; 
        for (Country otherCountry : countries) {
            if (otherCountry.id == K) continue; 

            if (otherCountry.gold > targetCountry.gold) {
                rank++;
            } else if (otherCountry.gold == targetCountry.gold && otherCountry.silver > targetCountry.silver) {
                rank++;
            } else if (otherCountry.gold == targetCountry.gold && otherCountry.silver == targetCountry.silver && otherCountry.bronze > targetCountry.bronze) {
                rank++;
            }
        }
        
        System.out.println(rank);
    }
}